### PR TITLE
Set destination branch

### PR DESCRIPTION
**Pull request by hhamid93**
Parameter for destination branch and support for new/existing branch was added in:
 
Tested this on nova-cluster. Testing was successful.

### CODE REVIEW

File: components/tekton/pipelines/create-cluster/create-cluster.yaml [hunk 1]
```diff
@@ -90,10 +95,10 @@ spec:
             git config --global user.name "redhat-testharness"
             git config --global user.email "redhat-partner-solutions@redhat.com"
             git config --global --add safe.directory /workspace/source
-            git checkout -b test-pipeline
+            git checkout -B "$(params.destination_branch)"
             git add $(params.hub_clusters_path)
             git commit -m "Copy folder from pipeline"
-            git push origin test-pipeline
+            git push -f origin "$(params.destination_branch)"
```
Comment (natifridman) [inline components/tekton/pipelines/create-cluster/create-cluster.yaml]: I don't think we need to force push

File: components/tekton/pipelines/create-cluster/create-cluster.yaml [hunk 2]
```diff
@@ -26,11 +26,16 @@ spec:
       name: hub_clusters_path
       type: string
       description: path for the clusters in the hub cluster repo
+<<<<<<< HEAD
```
Comment (natifridman) [inline components/tekton/pipelines/create-cluster/create-cluster.yaml]: Looks like leftovers from conflict

File: components/tekton/pipelines/create-cluster/create-cluster.yaml [hunk 3]
```diff
@@ -26,11 +26,16 @@ spec:
...
+    - name: destination_branch
+      type: string 
+=======
```
Comment (natifridman) [inline components/tekton/pipelines/create-cluster/create-cluster.yaml]: same

**Review by natifridman — CHANGES_REQUESTED** [review]
Looks good, left some comments

File: components/tekton/pipelines/create-cluster/create-cluster.yaml (hunk 1 above)
Comment (hhamid93) [reply components/tekton/pipelines/create-cluster/create-cluster.yaml]: so tried it without the -f option initially it errors out

File: components/tekton/pipelines/create-cluster/create-cluster.yaml (hunk 3 above)
Comment (hhamid93) [reply components/tekton/pipelines/create-cluster/create-cluster.yaml]: fixed

File: components/tekton/pipelines/create-cluster/create-cluster.yaml (hunk 2 above)
Comment (hhamid93) [reply components/tekton/pipelines/create-cluster/create-cluster.yaml]: fixed

File: components/tekton/pipelines/create-cluster/create-cluster.yaml (hunk 1 above)
Comment (natifridman) [reply components/tekton/pipelines/create-cluster/create-cluster.yaml]: I think we can start without force push and add this if we will see a need

Comment (natifridman) [reply components/tekton/pipelines/create-cluster/create-cluster.yaml]: I think I found the issue you had. You need to set `hub_git_revision` in the pipelinerun to the correct branch, the default is main

Comment (nocturnalastro) [reply components/tekton/pipelines/create-cluster/create-cluster.yaml]: Agreed you shouldn't need a force push is your just adding something to the head of the git log. 

A force push can override history which is not what we want here as it should be audit-able.

**Review by natifridman — APPROVED** [review]
LGTM

**Pull request merged**